pkg/kubelet/lifecycle/handlers.go

```diff
@@ -69,6 +70,9 @@ func (hr *HandlerRunner) Run(containerID kubecontainer.ContainerID, pod *v1.Pod,
 			glog.V(1).Infof(msg)
 		}
 		return msg, err
+	case handler.Sleep != nil:
+		time.Sleep(time.Duration(int64(handler.Sleep.Seconds) * int64(time.Second)))
```

Comment:
This will cause the calling code to block, potentially for a very long time.  That changes the semantics of how and where it can be used quite a lot.

If we're literally going to sleep here, we need to do a through examination of the entire callstack (all the potential callstacks, actually) to be sure it is safe to block.  I don't know Kubelet's internals well enough any more to say for sure, but a quick glance at callers seems to suggest that this would block `Kubelet.killPod` which DOES seem to be run from a dedicated goroutine.  So this may be OK but:

a) @yujuhong needs to confirm
b) This should be commented about here

Reply:
Hm, it does not make sense to sleep longer than the killPod signal. So we could potentially limit sleep to the deadline of the pod.

At least the `exec` hook has the same issue of potentially unbound runtime (well, up to killPod).

Reply:
ping @yujuhong